plugins/auth-common/src/guestKeys.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Holds the Guest User Identity Private Keys when allowGuestMode flag in Config is set.
+ *
+ * @remarks
+ *
+ *  `auth.allowGuestMode` flag: when using this flag, a `GuestUserIdentity` private key is added to the list of valid keys Users can authenticate with in the `DefaultClientAuthProvider`.
+ *  Guest Token only provided with the "development" and "test" process mode. Does not work with "production" process mode.
+ *
+ * @public
+ */
+export const GUEST_USER_PRIVATE_KEY: string =
+  '-----BEGIN PRIVATE KEY-----\n' +
+  'MIGHAgEAMBMGByqGSM49AgEGCCqGSM49AwEHBG0wawIBAQQgeK+2+vxWmdH3l5vS\n' +
+  'fWPXCPojy2dQ44hr3XoUN3pQ1KihRANCAATtkYfMKdb1cLkasRc87l7+Fu0BfNY3\n' +
+  'OMxEttX87GkP3+2Q6IzR1LSa9+h5APS781hKNPFlUQDnutzAuChCaP7Z\n' +
+  '-----END PRIVATE KEY-----\n';
```

Comment:
A thought here. Would it make sense to export two functions `generateGuestJwk` and `verifyGuestJwk` instead? Leaving the crypto details entirely hidden inside this common library. It wouldn't have to check config and whatnot; the policy of whether it applies or not can be left to the client and backend

Reply:
I can see this being useful if we move in the direction of the above comment where we continuously generate tokens.  I intentionally added the verification bit to the already existing TokenFactory.Authenticate method to make use of the already existing keyStore. Ii made sense to me to use what was already there, but yeah open to suggestions on this front as well.